plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -84,6 +84,78 @@ def unquarantine(self, agent_id: str) -> bool:
         failed = results_object.get("data").get("unquarantineAssets").get("results")[0].get("failed")
         return not failed
 
+    def quarantine_list(self, agent_hostname_list: List[str], advertisement_period: int) -> Tuple[List[str], List[str]]:
+        """
+        Quarantine an agent given a list of agent IDs
+
+        :param agent_hostname_list: List of agent hostnames to quarantine
+        :param advertisement_period: Amount of time, in seconds, to try to take the quarantine action
+
+        :return: Two lists containing asset ids for successful or unsuccessful quarantines
+        """
+
+        # Create empty lists for successful & unsuccessful
+        successful_quarantine = []
+        unsuccessful_quarantine = []
+
+        # Convert the original array of hostnames to IDs
+        agent_id_list = self._convert_hostnames_to_id(agent_hostname_list)
+
+        # For each agent ID in the list, perform quarantine
+        for agent in agent_id_list:
+            quarantine_payload = {
+                "query": "mutation( $orgID:String! $agentID:String! $advPeriod:Long! ) { quarantineAssets( "
+                "orgId:$orgID assetIds: [$agentID] input: {advertisementPeriod: $advPeriod} ) { results { "
+                "assetId failed } } }",
+                "variables": {
+                    "orgID": self.org_key,
+                    "agentID": agent,
+                    "advPeriod": advertisement_period,
+                },
+            }
+
+            results_object = self._post_payload(quarantine_payload)
+            asset_id = results_object.get("data", {}).get("quarantineAssets").get("results")[0].get("assetId")
+
+            if not results_object.get("data").get("quarantineAssets").get("results")[0].get("failed"):
```

Comment:
If there is an exception when posting the payload, is this handled here?

Reply:
No, the exception is handled in the `_post_payload` method